openedx/core/djangoapps/cookie_metadata/tests/test_cookie_name_change.py

```diff
@@ -0,0 +1,136 @@
+"""
+Test Modle to test CookieNameChange class
+"""
+from unittest.mock import Mock
+
+from django.test import TestCase
+
+
+from ..middleware import CookieNameChange
+
+
+class TestCookieNameChange(TestCase):
+    """
+    Test class for CookieNameChange
+    """
+
+    def test_cookie_swap(self):
+        """Check to make sure Middleware correctly swaps keys"""
+
+        mock_response = Mock()
+        middleware = CookieNameChange(mock_response)
+        mock_request = Mock()
+
+        old_value = "." * 100
+        extra_cookies = {
+            "_b": "." * 13,
+            "_c_": "." * 13,
+            "a.b": "." * 10,
+        }
+        old_dict = {
+            "a": old_value,
+        }
+        old_dict.update(extra_cookies)
+
+        mock_request.COOKIES = old_dict.copy()
+        expand_settings = {
+            "old_name": "a",
+            "new_name": "b",
+            "old_domain": "old_domain.com",
+        }
```

Comment:
nit: you can refactor this so that the common setup here is handled in `__init__` or `setUp`, which makes the differences between tests a little bit easier to reason about.